translate/v23tov30/v23tov30.go

```diff
@@ -447,3 +447,75 @@ func translateDirectories(dirs []old.Directory, m map[string]string) (ret []type
 	}
 	return
 }
+
+// RemoveDuplicateFilesAndUnits is a helper function that removes duplicated files/units from
+// spec v2 config, since neither spec v3 nor the translator function allow for duplicate file
+// entries in the config.
+// This functionality is not included in the Translate function, but may be useful in cases
+// where configuration has to be sanitized before translation.
+// For duplicates, it takes ordering into consideration by taking the file/unit contents from
+// the slice with the highest index value, which is assumed to be the latest revision.
+// Multiple filesystems are not considered, it is assumed there is only one.
+// File appendices are not considered.
```

Comment:
Shouldn't we fail in that case, rather than doing the wrong thing?

Reply:
added a fail condition for this